### ISSUE TITLE
Hotloading of tool mode is broken

### ISSUE THREAD

**Issue by StraToN** [godotengine/godot#8826]
**Operating system or device - Godot version:**
Mageia 5
Godot 2.2 custom build

**Issue description:**
Adding "tool" keyword to a script to enable it in the editor causes multiple errors in the console.
If you close Godot editor and restart it, tool mode does work directly.

Godot console (looping errors)
```
 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.

 res://scripts/fish_realistic.gd:65 - Invalid operands 'Nil' and 'float' in operator '/'.

 res://scripts/fish_realistic.gd:95 - Invalid type in built-in function 'cos'. Cannot convert argument 1 from Nil to float.

 res://scripts/fish_realistic.gd:120 - Invalid call. Nonexistent function 'length' in base 'Nil'.
```

**Steps to reproduce:**
Clone the project below (https://github.com/StraToN/campaines-fish.git)
Open scenes/3D/wandering_fishes.tscn.
Select a `fish*` node and edit the script (fish_realistic.gd)
Add the tool keyword on top. Save.
The console prints errors non-stop.
Remove the "tool" keyword. Save. 
The console stops printing errors as tool mode is disabled.
Re-add the "tool" keyword. Save. 
Restart Godot.
Reopen the scene.
Tool mode is enabled and working good.

**Link to minimal example project:**
https://github.com/StraToN/campaines-fish.git
Use the wandering_fishes.tscn scene located in scenes/3D

**Comment by kubecz3k** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about its current status here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
I've used `tool` mode every now and then in the past year and don't remember seeing this, so assuming fixed.
